library/primitive-types.lisp

```diff
@@ -40,6 +40,18 @@
     (lisp (List :a) ()
       cl:nil))
 
+  ;; Optional is an early type
```

Comment:
Add to comment: "The following act as constructors. It is functionally equivalent to the definition

```
(define-type (Optional :a)
  (Some :a)
  None)
```